agent/task_context.go

```diff
@@ -2,19 +2,41 @@ package agent
 
 import (
 	"context"
+	"sync"
 	"time"
 
 	"github.com/evergreen-ci/evergreen"
 	"github.com/evergreen-ci/evergreen/agent/command"
 	"github.com/evergreen-ci/evergreen/agent/internal"
+	"github.com/evergreen-ci/evergreen/agent/internal/client"
 	"github.com/evergreen-ci/evergreen/apimodels"
 	"github.com/evergreen-ci/evergreen/model"
 	"github.com/evergreen-ci/evergreen/model/patch"
 	"github.com/evergreen-ci/utility"
 	"github.com/mongodb/grip"
+	"github.com/mongodb/jasper"
 	"github.com/pkg/errors"
 )
 
+type taskContext struct {
+	currentCommand            command.Command
+	logger                    client.LoggerProducer
+	task                      client.TaskData
+	ranSetupGroup             bool
+	taskConfig                *internal.TaskConfig
+	timeout                   timeoutInfo
+	oomTracker                jasper.OOMTracker
+	traceID                   string
+	unsetFunctionVarsDisabled bool
+	// TODO (EVG-20289): see if taskDirectory can be replaced with
+	// TaskConfig.WorkDir.
```

Comment:
I intentionally left this for a follow-up PR to reduce the surface area of potential breakages from this change.